homeassistant/components/vicare/fan.py

```diff
@@ -0,0 +1,136 @@
+"""Viessmann ViCare ventilation device."""
+
+from __future__ import annotations
+
+from contextlib import suppress
+import logging
+
+from PyViCare.PyViCareDevice import Device as PyViCareDevice
+from PyViCare.PyViCareDeviceConfig import PyViCareDeviceConfig
+from PyViCare.PyViCareUtils import (
+    PyViCareInvalidDataError,
+    PyViCareNotSupportedFeatureError,
+    PyViCareRateLimitError,
+)
+from PyViCare.PyViCareVentilationDevice import (
+    VentilationDevice as PyViCareVentilationDevice,
+)
+from requests.exceptions import ConnectionError as RequestConnectionError
+
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    ordered_list_item_to_percentage,
+    percentage_to_ordered_list_item,
+)
+
+from .const import DEVICE_LIST, DOMAIN
+from .entity import ViCareEntity
+from .types import VentilationMode, VentilationProgram, ViCareDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+ORDERED_NAMED_FAN_SPEEDS = [
+    VentilationProgram.LEVEL_ONE,
+    VentilationProgram.LEVEL_TWO,
+    VentilationProgram.LEVEL_THREE,
+    VentilationProgram.LEVEL_FOUR,
+]
+
+
+def _build_entities(
+    device_list: list[ViCareDevice],
+) -> list[ViCareFan]:
+    """Create ViCare fan entities for a device."""
+
+    return [
+        ViCareFan(device.config, device.api, "ventilation")
+        for device in device_list
+        if isinstance(device.api, PyViCareVentilationDevice)
+    ]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the ViCare fan platform."""
+
+    device_list = hass.data[DOMAIN][config_entry.entry_id][DEVICE_LIST]
+
+    async_add_entities(
+        await hass.async_add_executor_job(
+            _build_entities,
+            device_list,
+        )
+    )
```

Comment:
Change like this if it's now OK to make the objects in the event loop
```suggestion
    async_add_entities(
        [
            ViCareFan(device.config, device.api, "ventilation")
            for device in device_list
            if isinstance(device.api, PyViCareVentilationDevice)
        ]
    )
```